src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -261,17 +261,16 @@ tf_compute_context::tf_compute_context() = default;
 tf_compute_context::~tf_compute_context() = default;
 
 size_t tf_compute_context::memory_budget() const {
-  // TODO: Returns 4GB as that makes sure default batch size is used. 
+  // TODO: Returns 4GB as that makes sure default batch size is used.
   // Do something that makes more sense like MPS later.
   return 4294967296lu;
 }
 
 std::vector<std::string> tf_compute_context::gpu_names() const {
-  pybind11::object gpu_devices;
   std::vector<std::string> gpu_device_names;
 
   call_pybind_function([&]() {
-    
+    pybind11::object gpu_devices;
```

Comment:
There is the change.

Reply:
Can you add it like `pybind11::object gpu_devices = tf_gpu_devices.attr("get_gpu_names")();`

Reply:
In addition, let's fix the other instances in this file. Let's only declare `pybind11::object` variables inside the lambda being passed to `call_pybind_function`